### PR TITLE
Add status to results

### PR DESCRIPTION
**Pull request by frankbuckley**

### CODE REVIEW

File: src/DSE.Open.Results/ResultHttpStatusCodeMapper.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+using System.Net;
+
+namespace DSE.Open.Results;
+
+public static class ResultHttpStatusCodeMapper
+{
+    public static HttpStatusCode ToHttpStatusCode(this ResultStatus status)
+    {
+        return status switch
```
Comment (rabuckley) [inline src/DSE.Open.Results/ResultHttpStatusCodeMapper.cs:12]: Unless we are going to start returning the most correct verbs, I'm not sure we need most of these. Probably just BadRequest, NotFound, Unauthorized, and Forbidden?

File: src/DSE.Open.Results/ResultStatus.cs [hunk 2]
```diff
@@ -0,0 +1,73 @@
+﻿// Copyright (c) Down Syndrome Education International and Contributors. All Rights Reserved.
+// Down Syndrome Education International and Contributors licence this file to you under the MIT license.
+
+namespace DSE.Open.Results;
+
+public enum ResultStatus
```
Comment (rabuckley) [inline src/DSE.Open.Results/ResultStatus.cs:6]: Unspecified maps to Ok iff no notifications error or above, else BadRequest?

**Review by rabuckley — APPROVED** [review]

File: src/DSE.Open.Results/ResultBuilder.cs [hunk 3]
```diff
@@ -9,12 +9,19 @@ namespace DSE.Open.Results;
 public abstract class ResultBuilder<TResult>
     where TResult : Result
 {
-    public ICollection<Notification> Notifications { get; } = new List<Notification>();
+    public virtual ResultStatus Status { get; set; }
```
Comment (rabuckley) [inline src/DSE.Open.Results/ResultBuilder.cs:12]: Why virtual?

File: src/DSE.Open.Results/ResultBuilder.cs [hunk 4]
```diff
@@ -9,12 +9,19 @@ namespace DSE.Open.Results;
...
 
     public virtual void MergeNotifications(Result result)
     {
         ArgumentNullException.ThrowIfNull(result);
 
+        if (Status == ResultStatus.Unspecified && result.Status != ResultStatus.Unspecified)
+        {
+            Status = result.Status;
+        }
+
```
Comment (rabuckley) [inline src/DSE.Open.Results/ResultBuilder.cs:24]: I don't like having ResultStatus being merged in a method called "MergeNotifications".

File: src/DSE.Open.Results/ResultHttpStatusCodeMapper.cs (hunk 1 above)
Comment (frankbuckley) [reply src/DSE.Open.Results/ResultHttpStatusCodeMapper.cs:12]: We should review this with respect to the APIs

File: src/DSE.Open.Results/ResultStatus.cs (hunk 2 above)
Comment (frankbuckley) [reply src/DSE.Open.Results/ResultStatus.cs:6]: I am in two minds about merging logic - e.g. if you have a `Found` but then get a `ServerError` from a later operation, presumably the error trumps the found.

Let's flow the change and see how it fits in actual use.

File: src/DSE.Open.Results/ResultBuilder.cs (hunk 3 above)
Comment (frankbuckley) [reply src/DSE.Open.Results/ResultBuilder.cs:12]: Was wondering if collection result builder might override with respect to `Found` if collection empty.

File: src/DSE.Open.Results/ResultBuilder.cs (hunk 4 above)
Comment (frankbuckley) [reply src/DSE.Open.Results/ResultBuilder.cs:24]: Yes, if we do not want to handle by default, we could provide `MergeStatus` and `MergeNotificationsAndStatus` methods.

**Pull request merged**